src/components/Link.tsx

```diff
@@ -0,0 +1,74 @@
+import React from "react";
+
+type LinkVariantProps = React.HTMLProps<HTMLAnchorElement> & {
+  variant?: "link";
+  href: string;
+};
+type NavVariantProps = React.HTMLProps<HTMLLIElement> & {
+  isActive: boolean;
+  variant: "nav";
+  onClick: () => void;
+};
+type Props = LinkVariantProps | NavVariantProps;
+
+/**
+ * A reusable link component with animated underline when the link is active.
+ *
+ * Has two variants:
+ *  1. "link" (default): renders an <a> tag with hover/focus styling.
+ *  2. "nav": renders a <button> with an onClick callback. Active state styling
+ *     is handled by an isActive prop, rather than directly by hover/focus.
+ */
+const Link = (props: Props) => {
+  const { variant, children } = props;
+
+  const getActiveStyle = (): string => {
+    if (variant === "nav") {
+      return props.isActive
+        ? "after:right-0 after:opacity-1"
+        : "after:right-full after:opacity-0";
+    } else {
+      return `
+        after:right-full
+        hover:after:right-0
+        focus:after:right-0
+
+        after:opacity-0
+        hover:after:opacity-100
+        focus:after:opacity-100
+      `;
+    }
+  };
```

Comment:
To clarify, I believe `Link` should look the same whether it's a `button` or `a` element under the hood. They differ in whether they're an interactive element, or whether they're used for navigation. At least, that's my general thinking. 😄 

Towards that, my suggestion would be,

```typescript
const getActiveStyle = (): string => `
  ${props.isActive
    ? "after:right-0 after:opacity-1"
    : "after:right-full after:opacity-0"
  }
  after:right-full
  hover:after:right-0
  focus:after:right-0
  after:opacity-0
  hover:after:opacity-100
  focus:after:opacity-100
`;
```

Or something to that effect. Does that make sense? Do let me know if you have questions, I'm all ears.

Reply:
So I guess I was thinking that they would have different conditions for when the underline appears—for the nav it's when we're on that section, but for the general link style it's just hover/focus, but I see what you're saying. 

If we want the nav styling for an active section to be in *addition* to hover/focus (which I agree is probably a good idea), I'll look into that probably tomorrow morning (just tried your example above and it looks like it removes the active nav bar underline, probably just a conflict between something below overriding it, should be a smaller thing to sort.

Thanks for all the testing/feedback :D

Reply:
Yeah no rush. 😄